scripts/keeper.ts

```diff
@@ -1,156 +1,127 @@
-    const { amount } = await l2TeleportGateway.estimateFee.flush([
-      encodedTargetDomain
-    ]);
-    const { transaction_hash } = await l2TeleportGateway.flush(
-      encodedTargetDomain,
-      { maxFee: amount * config.l2GasMultiplier }
-    );
-    await l2Signer.waitForTransaction(transaction_hash);
-    console.log("Success");
-  } else {
-    console.log("Flush not needed");
-  }
+  // eventually get targetDomains from File events
+  const targetDomains = ["GOERLI-MASTER-1"];
+  const domainsToFlush = [];
+
+  flushEvents.forEach(async (event) => {
+    const targetDomain = cairoShortStringToBytes32(BigNumber.from(event.args[1]));
+    domainsToFlush.push(targetDomain);
+  });
+  return domainsToFlush;
 }
 
-export async function finalizeFlush(config: Config) {
+export async function flush(config: Config) {
   const l1Signer = getL1Signer(config);
+  const l2Signer = getL2Signer(config);
 
-  const l1TeleportGateway = await getL1ContractAt<L1DAITeleportGateway>(
-    l1Signer,
-    "L1DAITeleportGateway",
-    config.l1TeleportGatewayAddress
-  );
-  const starknet = await getL1ContractAt<Starknet>(
-    l1Signer,
-    "Starknet",
-    config.starknetAddress
+  const l2TeleportGateway = await getL2ContractAt<l2_dai_teleport_gateway>(
+    l2Signer,
+    "l2_dai_teleport_gateway",
+    config.l2TeleportGatewayAddress
   );
 
-  const flushes = await flushesToBeFinalized(l1Signer, starknet, config);
-  for (let i = 1; i < flushes.length; i++) {
-    const flush = flushes[i];
-    console.log("Sending `finalizeFlush` transaction");
-    const tx = await l1TeleportGateway.finalizeFlush(
-      cairoShortStringToBytes32(flush.args.payload[1]),
-      flush.args.payload[2]
+  const domainsToFlush = await getDomainsToFlush(l2TeleportGateway, l1Signer, config);
+  domainsToFlush.forEach(async (targetDomain) => {
+    const [daiToFlushSplit] = await l2TeleportGateway.batched_dai_to_flush(
+      targetDomain
     );
-    await tx.wait();
-    console.log("Success");
-  }
-}
+    const daiToFlush = toUint(daiToFlushSplit);
+    console.log(`DAI to flush: ${daiToFlush}`);
 
-async function recentFlushTimestamp(
-  { teleportJoinAddress, sourceDomain, flushDelay }: Config,
-  l1Signer: Signer
-): Promise<number> {
-  const teleportJoin = await getL1ContractAt<TeleportJoin>(
-    l1Signer,
-    "TeleportJoin",
-    teleportJoinAddress
-  );
-  const settleFilter = teleportJoin.filters.Settle(l1String(sourceDomain));
-  const nearestBlock = await findNearestBlock(
-    l1Signer.provider,
-    Date.now() - 10 * flushDelay
-  );
-  const settleEvents = await teleportJoin.queryFilter(
-    settleFilter,
-    nearestBlock
-  );
-  const recentEvent = settleEvents[settleEvents.length - 1];
-  if (recentEvent) {
-    const block = await recentEvent.getBlock();
-    return block.timestamp;
-  } else {
-    return 0;
-  }
+    if (daiToFlush > config.flushMinimum) {
+      console.log("Sending `flush` transaction");
```

Comment:
In general logs should be more meaningful, in this case `targetDomain` and, `daiToFlush`, previous flush timestamp should be included.